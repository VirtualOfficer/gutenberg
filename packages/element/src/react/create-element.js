/**
 * External dependencies
 */
import { createElement } from 'react';

/**
 * Returns a new element of given type. Type can be either a string tag name or
 * another function which itself returns an element.
 *
 * @param {?(string|Function)} type     Tag name or element creator.
 * @param {Object}             props    Element properties, either attribute
 *                                       set to apply to node or values to
 *                                       pass through to element creator.
 * @param {...WPElement}       children Descendant elements.
 *
 * @return {WPElement} Element.
 */
export default createElement;
