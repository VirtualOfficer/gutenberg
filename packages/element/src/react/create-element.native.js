/**
 * External dependencies
 */
import { get, isString } from 'lodash';
import { createElement as createElementDefault } from 'react';
import {
	Circle,
	Defs,
	Ellipse,
	G,
	Line,
	Path,
	Polygon,
	TextPath,
	TSpan,
	Svg,
} from 'react-native-svg';

const htmlTagToNativeComponent = {
	circle: Circle,
	defs: Defs,
	ellipse: Ellipse,
	g: G,
	line: Line,
	path: Path,
	polygon: Polygon,
	svg: Svg,
	textPath: TextPath,
	tspan: TSpan,
	/*Polyline,
	Text,
	Rect,
	Use,
	Image,
	Symbol,
	LinearGradient,
	RadialGradient,
	Stop,
	ClipPath*/
};

const mapHTMLTagToNativeComponent = ( type ) => {
	if ( ! isString( type ) ) {
		return type;
	}

	return get( htmlTagToNativeComponent, [ type ], type );
};

/**
 * Returns a new element of given type. Type can be either a string tag name or
 * another function which itself returns an element.
 *
 * @param {?(string|Function)} type     Tag name or element creator.
 * @param {Object}             props    Element properties, either attribute
 *                                       set to apply to node or values to
 *                                       pass through to element creator.
 * @param {...WPElement}       children Descendant elements.
 *
 * @return {WPElement} Element.
 */
const createElement = ( type, props, ...children ) => {
	return createElementDefault(
		mapHTMLTagToNativeComponent( type ),
		props,
		...children
	);
};

export default createElement;
