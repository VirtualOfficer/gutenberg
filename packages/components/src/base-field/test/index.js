/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { BaseField } from '../index';

describe( 'props', () => {
	test( 'should render correctly', () => {
		const { container } = render( <BaseField /> );
		expect( container.firstChild ).toMatchSnapshot();
	} );
} );
