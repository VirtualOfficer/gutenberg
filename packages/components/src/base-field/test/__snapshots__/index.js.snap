// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  background: #fff;
  border-radius: 2px;
  border: 1px solid;
  border-color: #757575;
  box-shadow: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 13px;
  outline: none;
  padding: 0 8px;
  position: relative;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  width: 100%;
}

.emotion-0 > * + *:not(marquee) {
  margin-left: calc(4px * 2);
}

.emotion-0 > * {
  min-width: 0;
}

@media ( prefers-reduced-motion:reduce ) {
  .emotion-0 {
    -webkit-transition-duration: 0ms;
    transition-duration: 0ms;
  }
}

.emotion-0[disabled] {
  opacity: 0.6;
}

.emotion-0:hover {
  border-color: #757575;
}

.emotion-0:focus,
.emotion-0[data-focused='true'] {
  border-color: theme:var( --wp-admin-theme-color,#00669b);
  theme-dark10: var( --wp-admin-theme-color-darker-10,#007cba);
  box-shadow: 0 0 0,0.5px,[object Object];
}

<div
  class="components-flex components-base-field emotion-0 emotion-1 emotion-2"
  data-wp-c16t="true"
  data-wp-component="BaseField"
/>
`;
